nav2_mppi_controller/include/nav2_mppi_controller/motion_models.hpp

```diff
@@ -161,11 +144,18 @@ class AckermannMotionModel : public MotionModel
    */
   void applyConstraints(models::ControlSequence & control_sequence) override
   {
-    auto & vx = control_sequence.vx;
-    auto & wz = control_sequence.wz;
+    const auto vx_ptr = control_sequence.vx.data();
+    auto wz_ptr = control_sequence.wz.data();
+    int steps = control_sequence.vx.size();
+    for(int i = 0; i < steps; i++)
```

Comment:
Curious: if there's the ability to run a function over each element of an array, can that be used here to have the function do the clamping and applied to the array?

Reply:
Yes, there is "select" method and we have UnaryExpr methods also. But "select" method was slow compared to accessing using the for loop. I think SIMD instruction is not supported for clamping operation, hence overhead of Eigen templates is slowing down the operation. 
But, I'll confirm after comparing manual looping across raw pointers with Eigen::select and Eigen::UnaryExpr.

Reply:
Try using `fabs` instead so it doesn't use floats (which simd sucks at). Maybe its actually the `abs` rather than the `clamp` operation which should be fully templated - though some googling makes me think that `std::min(max, std::max(min, v));` is actually faster due to some of clamp's internal functions that we don't need for this case. You could try that here even without the `UnaryExpr` and possibly get a speed up

Reply:
You're right, `fabs` and using `min(upperlimit, max(lowerlimit))` is faster.

Reply:
yes, there are a few places. I'll replace them and check for improvement.